open-craft/problem-builder
# Refactor MentoringBlock to use StudioContainerWithNestedXBlocksMixin

Since we initially created ProblemBuilder, the StudioContainerWithNestedXBlocksMixin class has been added to the XBlock runtime. Since this is now the standardized way of dealing with nested XBlocks, we want to move MentoringBlock to use that structure and gain any existing benefits, as well as any benefits for the future.

**Dependencies**: None

**Screenshots**:
![screen shot 2016-09-28 at 7 38 21 am](https://cloud.githubusercontent.com/assets/7773758/18912111/be123f84-854e-11e6-9a7f-db753dc631f1.png)

![screen shot 2016-09-28 at 7 38 10 am](https://cloud.githubusercontent.com/assets/7773758/18912119/c3a40c98-854e-11e6-9732-9fd7682f04e7.png)

**Sandbox URL**:
- LMS : http://mentoring-block-refactor.opencraft.hosting
- Studio: http://studio-mentoring-block-refactor.opencraft.hosting

**Partner information**: Hosted on edx.org (Davidson College)

**Testing instructions**:

1. In either the sandbox or a devstack, verify that you are able to create a mentoring block.
2. Verify that subelements of the mentoring block can be created as expected.
3. Verify that you are able to drag and drop subelements to reorder them (not possible previously)

**Author notes and concerns**:

1. To prevent a circular import, we had to revert a performance enhancement change that imported MentoringBlock directly into the Questionnaire module. Because the QuestionnaireBlock is now used directly by the MentoringBlock, this resulted in a loop. Suggestions are welcome for a way we can keep the performance enhancement.

**Reviewers**
